client/layout/masterbar/logged-out.jsx

```diff
@@ -31,7 +31,7 @@ const MasterbarLoggedOut = ( { title, sectionName, translate } ) => (
 			}
 
 			{ 'login' !== sectionName
-			?	<Item url={ config( 'login_url' ) }>
+			?	<Item url={ config( 'login_url' ) + '?redirect_to=' + encodeURIComponent( window.location.href ) }>
```

Comment:
As discussed on Slack, in this context the `Sign in` link will not show, so I don't want to change that.